### PR TITLE
Brownfield New App Attach Tools VM - JCore

### PR DESCRIPTION
**Pull request by JCoreMS**
# Overview/Summary

Adding deployment code for MSIX App Attach Tools VM for packaging apps with MSIX.  Currently not using CARML but can updated if needed or required. 

## This PR fixes/adds/changes/removes

Deploys VM of OS choice with the following modifications
1. Installs MSIX App Attach
2. Installs PsfTooling App
3. Downloads and extracts MSIX Manager command line tool
4. Creates and installs self-signed certificate
5. Updates registry pre recommendations for App Attach VM and removes Edge and Windows welcome screens

### Breaking Changes

New add-on to brownfield

## Testing Evidence

Tested deployment in commercial and US Gov clouds

## As part of this Pull Request I have

- [X] Read the [Contribution Guide](https://github.com/Azure/avdaccelerator/blob/main/CONTRIBUTING.md) and ensured this PR is compliant with the guide
- [X] Ensured the resource API versions in `.bicep` file/s I am adding/editing are using the latest API version possible
- [X] Checked for duplicate [Pull Requests](https://github.com/Azure/avdaccelerator/pulls)
- [X] Associated it with relevant [GitHub Issues](https://github.com/Azure/avdaccelerator/issues)
- [X] *(AVD LZA Team Only)* Associated it with relevant [ADO Items](https://dev.azure.com/CSUSolEng/Accelerator%20-%20AVD/_backlogs/backlog/Accelerator%20-%20AVD%20Team/Features)
- [X] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/avdaccelerator/tree/main)
- [X] Performed testing and provided evidence.
- [X] Updated relevant and associated documentation (e.g. Contribution Guide, Module READMEs, Docs etc.)

### CODE REVIEW

File: workload/scripts/appAttachToolsVM/AppAttachVMConfig.ps1 [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+$AppAttachInstallFolder = "Microsoft.MSIXPackagingTool_1.2023.319.0_x64__8wekyb3d8bbwe"
+$PsfToolInstallFolder = "PsfTooling_6.3.0.0_x64__4y3s55xckzt36"
+
+# Create Log file for output and troublehsooting
+$Log = "C:\PostConfig.log"
+New-Item $Log
+Get-Date | Out-file $Log
+
+$Username = $ENV:COMPUTERNAME + '\' + $VMUserName
+$Password = ConvertTo-SecureString -String $VMUserPassword -AsPlainText -Force
```
[the remark this answers is not in the document]
Comment (JCoreMS) [reply workload/scripts/appAttachToolsVM/AppAttachVMConfig.ps1:34]: This is executed on the local VM after passing the credentials via the protected settings for the ARM/BICEP deployment.

File: workload/scripts/appAttachToolsVM/AppAttachVMConfig.ps1 [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+Invoke-WebRequest -URI $PsfToolPackageURL -OutFile "C:\MSIX\PsfTooling-x64.msix"
+If ($Error.Count -eq 0) { ".... COMPLETED!" | Out-File $Log -Append }
+Else { "-----ERROR-----> $Error" | Out-File $Log -Append; $Error.Clear() }
+
+"Enabling PSRemoting" | Out-file $Log -Append
+Enable-PSRemoting -Force
+If ($Error.Count -eq 0) { ".... COMPLETED!" | Out-File $Log -Append }
+Else { "-----ERROR-----> $Error" | Out-File $Log -Append; $Error.Clear() }
+
+Invoke-Command -ComputerName $ENV:COMPUTERNAME -Credential $VMCredential -ScriptBlock {
```
[the remark this answers is not in the document]
Comment (JCoreMS) [reply workload/scripts/appAttachToolsVM/AppAttachVMConfig.ps1:126]: Invoke-Command is used with the local computer only to allow install of Windows Apps within a user vs system context.

**Comment by JCoreMS** [comment]
@danycontre - this is now using CARML and ready to merge unless I need to fix anything else.

**Comment by danycontre** [comment]
Thanks @JCoreMS for your contribution.

**Pull request merged**